### ISSUE TITLE
Is there C to Forth compiler or research?

### ISSUE THREAD

**Issue by zhoupeng** [ForthHub/discussion#76]
compiler, papers or report on c to forth?

**Comment by pebhidecs** [comment]
On 28/10/2018 at 8:06 AM, "zhoupeng" <notifications@github.com> wrote:
>
>compiler, papers or report on c to forth?
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ForthHub/discussion/issues/76

There are such things around. See http://www.mpeforth.com/ for details of
their 'Sock-Puppet' which will handle c-code with the ease you can handle
Forth words. There might also be some papers in the past EuroForth
proceedings <http://www.complang.tuwien.ac.at/anton/euroforth/>.


Regards

Paul E. Bennett IEng MIET
Systems Engineer
Lunar Mission One Ambassador
-- 
********************************************************************
Paul E. Bennett IEng MIET.....
Forth based HIDECS Consultancy.............
Mob: +44 (0)7811-639972
Tel: Due to relocation - new number TBA. Please use Mobile.
Going Forth Safely ..... EBA. www.electric-boat-association.org.uk..
********************************************************************

**Comment by zhoupeng** [comment]
> On 28/10/2018 at 8:06 AM, "zhoupeng" ***@***.***> wrote: compiler, papers or report on c to forth? -- You are receiving this because you are subscribed to this thread. Reply to this email directly or view it on GitHub: #76
> There are such things around. See http://www.mpeforth.com/ for details of their 'Sock-Puppet' which will handle c-code with the ease you can handle Forth words. There might also be some papers in the past EuroForth proceedings <http://www.complang.tuwien.ac.at/anton/euroforth/>. Regards Paul E. Bennett IEng MIET Systems Engineer Lunar Mission One Ambassador -- 
Thank you Mr Paul very much, it's helplful.
but I think Sock-Puppet give solution on how to call C libary from Forth, rather than compile C source to Forth directly, if my understanding wrong pls correct me.

I find a report https://github.com/daniRiesgo/C-to-Forth-compiler/blob/master/report.pdf, but unfortunately, it is not written in English.

Thanks,
Best,

**Comment by pebhidecs** [comment]
>> There are such things around. See http://www.mpeforth.com/ for 
>>details of their 'Sock-Puppet' which will handle c-code with the 
>>ease you can handle Forth words. There might also be some papers 
>>in the past EuroForth proceedings 
>><http://www.complang.tuwien.ac.at/anton/euroforth/>.
>
>Thank you Mr Paul very much, it's helplful.
>but I think Sock-Puppet give solution on how to call C libary from 
>Forth, rather than compile C source to Forth directly, if my 
>understanding wrong pls correct me.
>
>I find a report https://github.com/daniRiesgo/C-to-Forth-
>compiler/blob/master/report.pdf, but unfortunately, it is not 
>written in English.

Yes, the C has to be compiled first, then Sock-Puppet acts like the
linker for the C routines.

The one you found looks like Spanish. I guess some time with a translator
package might yield most of the sense out of it.

Regards

Paul E. Bennett IEng MIET
Systems Engineer
Lunar Mission One Ambassador
-- 
********************************************************************
Paul E. Bennett IEng MIET.....
Forth based HIDECS Consultancy.............
Mob: +44 (0)7811-639972
Tel: Due to relocation - new number TBA. Please use Mobile.
Going Forth Safely ..... EBA. www.electric-boat-association.org.uk..
********************************************************************

**Comment by zhoupeng** [comment]
> > There are such things around. See http://www.mpeforth.com/ for >details of their 'Sock-Puppet' which will handle c-code with the >ease you can handle Forth words. There might also be some papers >in the past EuroForth proceedings ><http://www.complang.tuwien.ac.at/anton/euroforth/>. Thank you Mr Paul very much, it's helplful. but I think Sock-Puppet give solution on how to call C libary from Forth, rather than compile C source to Forth directly, if my understanding wrong pls correct me. I find a report https://github.com/daniRiesgo/C-to-Forth- compiler/blob/master/report.pdf, but unfortunately, it is not written in English.
> Yes, the C has to be compiled first, then Sock-Puppet acts like the linker for the C routines. The one you found looks like Spanish. I guess some time with a translator package might yield most of the sense out of it. Regards Paul E. Bennett IEng MIET Systems Engineer Lunar Mission One Ambassador -- ******************************************************************** Paul E. Bennett IEng MIET..... Forth based HIDECS Consultancy............. Mob: +44 (0)7811-639972 Tel: Due to relocation - new number TBA. Please use Mobile. Going Forth Safely ..... EBA. [www.electric-boat-association.org.uk](http://www.electric-boat-association.org.uk).. ********************************************************************

It's good, thank you very much.

**Comment by TG9541** [comment]
You might want to have a look at the [ELVM Compiler Infrastructure](https://github.com/shinh/elvm), even if it looks more like a proof of concept. The (very simple) Forth backend is [here](https://github.com/shinh/elvm/blob/master/target/forth.c).

**Comment by spelc** [comment]
There's a community version from long ago at:
  https://www.mpeforth.com/resource-links/downloads/
Look for C2Forth.

**Comment by cwpjr** [comment]
I'm in retirement on ss in America. I still intend to save for your modern
product Stephan,

**Comment by spelc** [comment]
The code link at
  https://www.mpeforth.com/resource-links/downloads/
disappeared a while back but has now been restored.

**Comment by mak4444** [comment]
https://github.com/mak4444/dbgspf/blob/main/CC4.f

**Comment by CCurl** [comment]
https://github.com/yesco/c2forth
